### PR TITLE
feat: add binary_to_decimal.cpp

### PR DESCRIPTION
**Pull request by vishwakrishnan02**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
This is an algorithm to convert a binary string to decimal format. 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1908"></a>

### CODE REVIEW

File: others/binary_to_decimal.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * Time Complexity: O(n)
+ * Space Complexity: O(n)
+ *
+ * @author [Vishwa Krishnan](https://github.com/vishwakrishnan02)
+ */
+
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <algorithm>
```
Comment (Panquesito7) [inline others/binary_to_decimal.cpp:15]: Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```

File: others/binary_to_decimal.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    //Input: 00000001 | Output: 1
+    assert(others::binary_to_decimal::binary_to_decimal("00000001") == 1);
+
+    std::cout << "Test cases passed." << std::endl;
+
+}
+
+int main()
+{
+    test(); //Runs all the self-implemented test cases.
```
Comment (Panquesito7) [inline others/binary_to_decimal.cpp:87]:
```suggestion
    test();  // run self-test implementations
```

File: others/binary_to_decimal.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    assert(others::binary_to_decimal::binary_to_decimal("11111111") == 255);
+
+    //Input: 00000001 | Output: 1
+    assert(others::binary_to_decimal::binary_to_decimal("00000001") == 1);
+
+    std::cout << "Test cases passed." << std::endl;
+
+}
+
+int main()
```
Comment (Panquesito7) [inline others/binary_to_decimal.cpp:85]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main()
```

File: others/binary_to_decimal.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                i++;
+            }
+
+            return decimal;
+        }
+    }
+}
+
+
+static void test() {
```
Comment (Panquesito7) [inline others/binary_to_decimal.cpp:61]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns 0 on exit
 */
static void test() {
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! Thank you. 🎉
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/vishwakrishnan02/C-Plus-Plus/actions

**Review by vishwakrishnan02 — COMMENTED** [review]
All changes have been made!

**Pull request closed without merging**